java/flight/pom.xml

```diff
@@ -20,7 +20,7 @@
   <packaging>jar</packaging>
 
   <properties>
-    <dep.grpc.version>1.14.0</dep.grpc.version>
+    <dep.grpc.version>1.24.0</dep.grpc.version>
```

Comment:
This is a pretty big version jump...do we have any policy for bumping dependency versions?

Reply:
We don't really have one for experimental things. Do you think we should avoid the bump?

Reply:
I think this is fine, with the caveat that technically, gRPC 1.24 requires Protobuf 3.7, not the 3.5 version we have pinned. (We're stuck on an older gRPC, but I don't think anything in Flight is particularly version-specific beyond the serialization tricks in ArrowMessage.)